src/game/city.go

```diff
@@ -262,3 +263,36 @@ func upgradeBuilding(response http.ResponseWriter, request *http.Request) {
 
 	status = true
 }
+
+func updateName(response http.ResponseWriter, request *http.Request) {
+	fmt.Println("Received request to /cities/updateName")
+	vars := mux.Vars(request)
+	sessionId := vars["session_id"]
+	status := false
+
+	defer func() {
+		json.NewEncoder(response).Encode(Status{Status: status})
+	}()
+
+	var newName string
+	err := json.NewDecoder(request.Body).Decode(&newName)
+
+	if err != nil {
+		return
+	}
+
+	query := fmt.Sprintf("UPDATE Cities SET city_name ='%s' FROM Sessions WHERE player_id=(SELECT player_id FROM Sessions NATURAL JOIN Cities JOIN Sessions ON city_owner=player_id WHERE session_id='%s')", newName, sessionId)
+	result, err := database.Execute(query)
+
+	if err != nil {
+		fmt.Println("There is an error")
```

Comment:
We want to return if an error exists instead of printing something, this is the only way that the defer statement will immediately get called to return false back to the client